### ISSUE TITLE
method for get + del 

### ISSUE THREAD

**Issue by KrishnaPG** [node-cache/node-cache#159]
For storing one-time reference items, such as OTP, it would be useful to have a method, say `take`, that would return the object and also delete it from the cache (so that any further `get` will result in `undefined`).

Something like below:
````
function take(key) {
  const val = this.get(key);
  if(val) this.delete(key);
  return val;
}
````

While this can be easily done at the app-level, for stronger semantics it would be nice to have in the API itself, if possible.

**Comment by sujeshthekkepatt** [comment]
I think you can achieve the same with `ttl`. Store for a particular time and then automatically invalidate it. Otp's are also after a particular time frame not be re-used, I think. What do you think ?

**Comment by KrishnaPG** [comment]
Single-use tokens (such as OTPs) cannot be used under two circumstances:

1. time expired,
2. already used

TTL solves the first problem. This `take` method aims to solve the second problem. The token could be active (TTL not expired), but if it is already "touched" (`get()` called for verification), then just delete it, so that no replay attack is possible.

**Comment by sujeshthekkepatt** [comment]
It's a very specific use case I think. You can however make use of the current methods and implement your use case. Like `get` in combination with `del`.

**Comment by erdii** [comment]
Hey @KrishnaPG this sounds like a decent use-case.

PR welcome 😊

**Comment by KrishnaPG** [comment]
Thank you. PR https://github.com/node-cache/node-cache/pull/160

**Comment by erdii** [comment]
Thank you! :+1: pr was merged and published in v5.0.2!
